src/LayoutManager/LayoutStrategies/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { Point } from '../../Point';
+import type { Group } from '../../shapes/Group';
+import type { FabricObject } from '../../shapes/Object/FabricObject';
+import type { TMat2D } from '../../typedefs';
+import { cos } from '../../util/misc/cos';
+import { multiplyTransformMatrices, qrDecompose } from '../../util/misc/matrix';
+import { calcPlaneChangeMatrix } from '../../util/misc/planeChange';
+import { degreesToRadians } from '../../util/misc/radiansDegreesConversion';
+import { sin } from '../../util/misc/sin';
+
+/**
+ * @todo this is very buggy => refactor in redo coords
+ */
+export const getObjectSizeVector = (
+  object: FabricObject,
+  /**
+   * account for selected objects in active selection
+   */
+  planeChange?: TMat2D
+) => {
+  const decomposition = planeChange
+    ? qrDecompose(
+        multiplyTransformMatrices(planeChange, object.calcOwnMatrix())
+      )
+    : undefined;
+  const sizeVector = object
+    ._getTransformedDimensions(decomposition)
+    .scalarDivide(2);
+  if (decomposition?.angle ?? object.angle) {
+    const rad = degreesToRadians(object.angle),
+      sine = Math.abs(sin(rad)),
+      cosine = Math.abs(cos(rad)),
+      rx = sizeVector.x * cosine + sizeVector.y * sine,
+      ry = sizeVector.x * sine + sizeVector.y * cosine;
+    return new Point(rx, ry);
+  }
+  return sizeVector;
+};
```

Comment:
is this a way to find the object bounding box aligned to group axis? or what does it do?

Reply:
caclulating objects size in group plane and taking into account the fact that an object can be selected, so it may need a plane change matrix

Reply:
Resurfacing #8767... this is here because that PR/concept should redo this entire section which has bugs with rotation of selected objects

Reply:
we shouldn't do this. _getTransformedDimensions wasn't meant to extend in this way and accepting skew values was already wrong.
This is a copy of code we have also somewhere else.
So this is calculating the object size but with which kind of rotation? is this aligned with the group bounding box?